Source/Dafny/Dafny.atg

```diff
@@ -434,6 +434,16 @@ bool IsNotEndOfCase() {
   return la.kind != _EOF && la.kind != _rbrace && la.kind != _case;
 }
 
+/* Checks if the expression starts with "var" (i.e. a let expression with a lhs) */
+bool IsLetExpr(){
+   scanner.ResetPeek();
+  Token x;
+  x = scanner.Peek();
```

Comment:
These two can be on one line, and I also suggest replacing the explicit type `Token` with just `var`.